sys/include/xtimer.h

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup  sys_xtimer Timers
+ * @ingroup   sys
+ * @brief     Provides a high level timer module to register
+ *            timers, get current system time, and let a thread sleep for
+ *            a certain amount of time.
+ *
+ * The implementation takes one low-level timer that is supposed to run at 1MHz
+ * speed and multiplexes it.
+ *
+ * Insertion and removal of timers has O(n) complexity with (n) being the
+ * number of active timers.  The reason for this is that multiplexing is
```

Comment:
@kaspar030, you're right discussing conceptional stuff at the very end of the work is annoying. Sorry for that. (Ask authors at IETF how many times fundamental critics are given when a documents proceed to working group last call...) However, since this module is gonna intended to be the default timer system for all RIOT applications from now on, breaking real-time guarantees with ope's eyes open seems to be the wrong choice to me.
But RIOT is community driven effort and two maintainers seem not to object this, I will not object.